core/math/vector2i.h

```diff
@@ -85,6 +85,14 @@ struct _NO_DISCARD_ Vector2i {
 		return Vector2i(MAX(x, p_vector2i.x), MAX(y, p_vector2i.y));
 	}
 
+	double distance_to(const Vector2i &p_vector2i) const {
+		return Math::sqrt((double)((x - p_vector2i.x) * (x - p_vector2i.x) + (y - p_vector2i.y) * (y - p_vector2i.y)));
+	}
+
+	int64_t distance_squared_to(const Vector2i &p_vector2i) const {
+		return (x - p_vector2i.x) * (x - p_vector2i.x) + (y - p_vector2i.y) * (y - p_vector2i.y);
+	}
+
```

Comment:
If we decided to go ahead with these functions, imo it would be best to copy technique from `Vector3i::length_squared()` and `Vector3i::length()` and create equivalent functions in `Vector2i`, these are better written imo. I suspect yours may be susceptible to overflow - I haven't tried them to see the assembly, but they are confusingly written either way, not making clear what you expect the compiler to do.

The only advantage to rewriting _not_ using `length()` and `length_squared()` seems to be to avoid overflow in the subtraction by using 64 bit. But then this operation will be slower, and the main risk of overflow is in the multiplication, so using 32 bit subtraction would probably be fine.

Reply:
In short what you're saying is to reimplement these methods for `Vector2i` as such:

```c++
double Vector2i::distance_to(const Vector2i &p_to) const {
	return (p_to - *this).length();
}

int64_t Vector2i::distance_squared_to(const Vector2i &p_to) const {
	return (p_to - *this).length_squared();
}
```

or am I missing something?

Reply:
Yes this is correct.

Reply:
Shouldn't we also refactor `Vector2` in this case?

https://github.com/godotengine/godot/blob/b1371806ad3907c009458ea939bd4b810f9deb21/core/math/vector2.cpp#L72-L78

As for similar PRs, the only one that comes to mind is #47653, but that one was scrapped for a different reason.

Reply:
Updated in https://github.com/godotengine/godot/pull/83163/commits/718c7c0858bb425ab119e164c7306095da34ef8c.